### PR TITLE
Extended XCM Test Cases

### PR DESCRIPTION
**Pull request by zktony**
## Describe your changes
Extended XCM Test Cases.

### CODE REVIEW

File: pallets/xcm-helper/src/lib.rs [hunk 1]
```diff
@@ -407,31 +411,56 @@ pub mod pallet {
...
+					asset_id.into(),
+					amount,
+					&account,
+					pallet_account.clone(),
+					1u128,
+					pallet_account.clone(),
+				)
+				.is_err()
+				{
+					log::error!(target:"xcm-helper","Deposit Failed");
```
Comment (Gauthamastro) [inline pallets/xcm-helper/src/lib.rs:439]: Deposit a failure event here

File: pallets/xcm-helper/src/lib.rs [hunk 2]
```diff
@@ -460,14 +495,16 @@ pub mod pallet {
...
 			let MultiAsset { id, fun } = asset;
 			let from =
 				T::AccountIdConvert::convert_location(from).ok_or(XcmError::FailedToDecode)?;
 			let to = T::AccountIdConvert::convert_location(to).ok_or(XcmError::FailedToDecode)?;
 			let amount: u128 = Self::get_amount(fun).ok_or(XcmError::Trap(101))?;
 			let asset_id = Self::generate_asset_id_for_parachain(*id);
-			Self::resolve_transfer(asset_id.into(), &from, &to, amount)
-				.map_err(|_| XcmError::Trap(102))?;
+
+			// Self::resolve_transfer(asset_id.into(), &from, &to, amount)
```
Comment (Gauthamastro) [inline pallets/xcm-helper/src/lib.rs]: Remove comments

File: pallets/xcm-helper/src/lib.rs [hunk 3]
```diff
@@ -460,14 +495,16 @@ pub mod pallet {
 			to: &MultiLocation,
 			_context: &XcmContext,
 		) -> sp_std::result::Result<Assets, XcmError> {
+			//panic!("Transfer is called from {:?} to {:?} with asset {:?}", from, to, asset);
```
Comment (Gauthamastro) [inline pallets/xcm-helper/src/lib.rs]: Same here

File: pallets/xcm-helper/src/lib.rs [hunk 4]
```diff
@@ -678,21 +727,20 @@ pub mod pallet {
...
+													.into_account_truncating(),
+											)
+											.into(),
+											Box::new(vec![multi_asset, fee_multi_asset].into()),
+											1,
+											Box::new(destination.clone()),
+											cumulus_primitives_core::WeightLimit::Unlimited,
+										) {
 										failed_withdrawal.push(withdrawal.clone());
-										log::error!(target:"xcm-helper","Withdrawal failed: Not able to make xcm calls");
```
Comment (Gauthamastro) [inline pallets/xcm-helper/src/lib.rs:695]: Deposit failure event

File: pallets/xcm-helper/src/lib.rs [hunk 5]
```diff
@@ -708,8 +756,40 @@ pub mod pallet {
 								};
 								let pallet_account: T::AccountId =
 									T::AssetHandlerPalletId::get().into_account_truncating();
+								let destination_location = match destination {
+									VersionedMultiLocation::V3(location) => location,
+									_ => {
+										failed_withdrawal.push(withdrawal);
+										log::error!(target:"xcm-helper","Withdrawal failed: Not able to decode destination");
```
Comment (Gauthamastro) [inline pallets/xcm-helper/src/lib.rs:772]: Deposit event

File: pallets/xcm-helper/src/lib.rs [hunk 6]
```diff
@@ -708,8 +756,40 @@ pub mod pallet {
...
+										withdrawal.asset_id.into(),
+										withdrawal.amount,
+										&soverign_account,
+										pallet_account.clone(),
+										1u128,
+										pallet_account.clone(),
+									)
+									.is_err()
+									{
+										failed_withdrawal.push(withdrawal.clone());
```
Comment (Gauthamastro) [inline pallets/xcm-helper/src/lib.rs:797]: Failure event here

File: pallets/xcm-helper/src/lib.rs [hunk 7]
```diff
@@ -722,20 +802,20 @@ pub mod pallet {
 									failed_withdrawal.push(withdrawal.clone());
 									log::error!(target:"xcm-helper","Withdrawal failed: Not able to mint token");
```
Comment (Gauthamastro) [inline pallets/xcm-helper/src/lib.rs:815]: Failure event here

File: pallets/xcm-helper/src/lib.rs [hunk 8]
```diff
@@ -722,20 +802,20 @@ pub mod pallet {
...
+												.into_account_truncating(),
+										)
+										.into(),
+										Box::new(multi_asset.into()),
+										0,
+										Box::new(destination.clone()),
+										cumulus_primitives_core::WeightLimit::Unlimited,
+									) {
 									failed_withdrawal.push(withdrawal.clone());
-									log::error!(target:"xcm-helper","Withdrawal failed: Not able to make xcm calls");
```
Comment (Gauthamastro) [inline pallets/xcm-helper/src/lib.rs:738]: Failure event here

File: pallets/xcm-helper/src/lib.rs [hunk 9]
```diff
@@ -744,6 +824,7 @@ pub mod pallet {
 									))
 								}
 							} else {
+								log::error!(target:"xcm-helper","Withdrawal failed: Not able to handle dest");
```
Comment (Gauthamastro) [inline pallets/xcm-helper/src/lib.rs:840]: Failure event here

File: pallets/xcm-helper/src/lib.rs [hunk 10]
```diff
@@ -651,7 +692,15 @@ pub mod pallet {
 									};
 									let pallet_account: T::AccountId =
 										T::AssetHandlerPalletId::get().into_account_truncating();
-									if Self::resolver_deposit(
+									let destination_location = match destination {
+										VersionedMultiLocation::V3(location) => location,
+										_ => {
+											failed_withdrawal.push(withdrawal);
+											log::error!(target:"xcm-helper","Withdrawal failed: Not able to decode destination");
```
Comment (Gauthamastro) [inline pallets/xcm-helper/src/lib.rs]: Failure event here

File: polkadex-xcm-simulator/Cargo.toml [hunk 11]
```diff
@@ -28,6 +29,23 @@ polkadot-core-primitives = { git = "https://github.com/paritytech/polkadot-sdk",
...
+orml-traits = { git = "https://github.com/Polkadex-Substrate/orml-1.1.0.git", branch = "master" }
+
+#Polkadex Pallets
+thea-primitives = { workspace = true }
+polkadex-primitives = { workspace = true }
+xcm-helper = { path = "../pallets/xcm-helper" }
+thea-message-handler = { path = "../pallets/thea-message-handler" }
+thea = { path = "../pallets/thea" }
+smallvec = "1.13.1"
+#polkadot-runtime-common = { workspace = true, default-features = false }
```
Comment (Gauthamastro) [inline polkadex-xcm-simulator/Cargo.toml]: Remove this

**Review by Gauthamastro — CHANGES_REQUESTED** [review]
@zktony Where are the test cases in simulator for sending PDEX from mainnet to parachain

**Comment by zktony** [comment]
> @zktony Where are the test cases in simulator for sending PDEX from mainnet to parachain
https://github.com/Polkadex-Substrate/Polkadex/blob/95832aa4bfe9aeb7a498d8edff29b9ef20e7315e/polkadex-xcm-simulator/src/lib.rs#L524

**Comment by Gauthamastro** [comment]
> > @zktony Where are the test cases in simulator for sending PDEX from mainnet to parachain
> > https://github.com/Polkadex-Substrate/Polkadex/blob/95832aa4bfe9aeb7a498d8edff29b9ef20e7315e/polkadex-xcm-simulator/src/lib.rs#L524

Update the name, from non-native to native. Its misleading

File: pallets/xcm-helper/src/lib.rs (hunk 1 above)
Comment (serhii-temchenko) [reply pallets/xcm-helper/src/lib.rs:439]: Still not resolved.

File: pallets/xcm-helper/src/lib.rs (hunk 6 above)
Comment (serhii-temchenko) [reply pallets/xcm-helper/src/lib.rs:797]: Still not resolved.

**Review by serhii-temchenko — CHANGES_REQUESTED** [review]

**Review by Gauthamastro — CHANGES_REQUESTED** [review]
Two failure events are missing

**Review by Gauthamastro — APPROVED** [review]

**Review by serhii-temchenko — APPROVED** [review]

**Pull request merged**